pkg/download/protocol.go

```diff
@@ -89,7 +89,7 @@ func (p *protocol) List(ctx context.Context, mod string) ([]string, error) {
 
 func (p *protocol) Latest(ctx context.Context, mod string) (*storage.RevInfo, error) {
 	const op errors.Op = "protocol.Latest"
-	ctx, span := observ.StartSpan(ctx, op.String())
+	_, span := observ.StartSpan(ctx, op.String())
```

Comment:
I am not sure why this change is required. It decreases readbility and also introduces the possibility of errors when a new function taking a function is called.

Reply:
Agree with manu, we need to shadow ctx here because it's super easy to miss this little detail and have a headache if we pass `ctx` to a new function and the Traces are not being recorded.

Reply:
👍 on this one. I think we should keep it. price of not having a correct relationship between spans is higher than this.